packages/block-library/src/page-list-item/edit.js

```diff
@@ -30,10 +30,95 @@ function useFrontPageId() {
 	}, [] );
 }
 
+/**
+ * Determine the colors for a menu.
+ *
+ * Order of priority is:
+ * 1: Overlay custom colors (if submenu)
+ * 2: Overlay theme colors (if submenu)
+ * 3: Custom colors
+ * 4: Theme colors
+ * 5: Global styles
+ *
+ * @param {Object}  context
+ * @param {boolean} isSubMenu
+ */
+function getColors( context, isSubMenu ) {
```

Comment:
We sholud share this code from a canonical location as it's liable to produce bugs. We can do that here on in a followup which we should create an Issue for now.